New Features for 2025-06-01-preview API version

melissamserv opened the pull request:
New Features for **2025-06-01-preview** API version:

- Support for URIs and IP addresses in SAN properties:
  Add **uris** and **ipAddresses** attributes to Subject Alternative Names in certificate requests.

- PFX to PEM conversion in Get Certificate:
  Add the **outContentType** query parameter in the Get Certificate operation to specify the format in which the certificate will be returned.

- Include the version of the previous certificate in secret responses:
  Add the **previousVersion** attribute to the SecretBundle in secret responses.

melissamserv committed:
Remove unnecessary backticks in v2025_06_01_preview enum

melissamserv committed:
Add uris and ipAddresses attributes to SubjectAlternativeNames

melissamserv committed:
Add previousVersion attribute

melissamserv committed:
Add outContentType parameter

melissamserv committed:
Improve descriptions of the previousVersion attribute and outContentType parameter

heaths commented on specification/keyvault/Security.KeyVault.Secrets/models.tsp:
```diff
@@ -175,6 +175,12 @@ model SecretBundle {
   managed?: boolean;
+
+  /**
+   * The version of the previous certificate, if applicable. Applies only to certificates created after June 1, 2025. Certificates created before this date are not retroactively updated.
```
Could you clarify if this is a URI or just the version string (a GUID, though don't mention that) itself?

melissamserv replied:
Yes, it's a random GUID without hyphens. I didn't specify it because we don't specify this for the `secret-version` attribute either. Do you want me to update both?

heaths replied:
You should document that it's just the version. I put the bit about GUID in parenthesis only as an aside. No, you shouldn't document that specifically. But as written, should a caller expect a URI or something else back? If it's just the version string, I would just say that.

melissamserv replied:
It's only the version string, not the complete URI, otherwise it will be confused with the "id" attribute. - Discussed offline, I'll keep the description since that's how our secret-version path segment is currently worded.

heaths commented on specification/keyvault/Security.KeyVault.Secrets/routes.tsp:
```diff
@@ -113,6 +113,13 @@ op getSecret is KeyVaultOperation<
     secretVersion: string;
+
+    /**
+     * The media type (MIME type) of the certificate. If specified, converts the certificate content to the requested format. Currently, only PFX to PEM conversion is supported. If not specified, the certificate is returned in its original format without conversion.
```
Do you 400 otherwise? Might be good to clarify so devs know what to expect.

melissamserv replied:
No, otherwise we ignore the value, we only support the case when they specify the **outContenType** as PEM, since we only allow the PEM conversion.

melissamserv committed:
Fix TSP validation for SecurityDomain

Add new 2025-06-01-preview API version to SecurityDomain spec
Copy TSP SecurityDomain examples from 7.6 to 2025-06-01-preview and update the API version

melissamserv committed:
Improve description of outContentType parameter

heaths commented on specification/keyvault/Security.KeyVault.Secrets/routes.tsp:
```diff
@@ -115,7 +115,7 @@ op getSecret is KeyVaultOperation<
 
     /**
-     * The media type (MIME type) of the certificate. If specified, converts the certificate content to the requested format. Currently, only PFX to PEM conversion is supported. If not specified, the certificate is returned in its original format without conversion.
+     * The media type (MIME type) of the certificate. If specified, converts the certificate content to the requested format. Currently, only PFX to PEM conversion is supported. If not specified or unsupported, the certificate is returned in its original format without conversion.
```
Why not error if not supported? That's what an API should do.

melissamserv committed:
Fix outContentType parameter description

laiapat committed:
Update tspconfig.yaml

laiapat committed:
Re-run checks

heaths commented:
/azp run

melissamserv commented:
@microsoft-github-policy-service agree company="Microsoft"

> @melissamserv please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement
 @microsoft-github-policy-service agree company="Microsoft"
